test/document.test.js

```diff
@@ -14225,6 +14225,47 @@ describe('document', function() {
     assert.strictEqual(duplicateKeyError.message, 'Email must be unique');
     assert.strictEqual(duplicateKeyError.cause.code, 11000);
   });
+
+  it('supports global transforms per schematype (gh-15084)', async function () {
+    class SchemaCustomType extends mongoose.SchemaType {
+      static schemaName = 'CustomType';
+
+      constructor(key, options) {
+        super(key, options, 'CustomType');
+      }
+
+      cast(value) {
+        if (value === null) return null;
+        return new CustomType(value);
+      }
+    }
+
+    class CustomType {
+      constructor(value) {
+        this.value = value;
+      }
+    }
+
+    mongoose.Schema.Types.CustomType = SchemaCustomType;
+
+    const Model = db.model(
+      'Test',
+      new mongoose.Schema({
+        value: { type: mongoose.Schema.Types.CustomType },
```

Comment:
```suggestion
        value: { type: mongoose.Schema.Types.CustomType }
```

Reply:
Fixed in [08113ef](https://github.com/Automattic/mongoose/pull/15163/commits/08113efa08ccec25265523a92a8075a48aabf057)